### ISSUE TITLE
`standardize_response` is awkward

### ISSUE THREAD

**Issue by aaron-suarez** [OperationCode/resources_api#97]
I like the idea of having a function to standardize all of the responses so we can respond in certain ways without having to think too hard about how to structure the response while maintaining a consistent API experience.

However, the interface for the current implementation is pretty awful. At the very least, it should use keyword arguments so that it's obvious which parameter is which when you're reading through the code. But, perhaps with a little thought, a solution that's a bit less clunky and more readable can be designed.

I'm open to suggestions about how this might look.

**Comment by mildocjr** [comment]
It seems to be a typical format, however, something I do is pass a dictionary in the format `{'data': data, 'errors': [errors]}` to a `payload` argument and let the browser handle the OK status.  Then you could pull out each of these in the `standardize_response` function, and determine an appropriate error code which could be overridden if needed as a named argument.

General format would be:
```python
def standarize_response(payload, status_code=200):
    # pull out possible values from payload or set to some default value 
    data = payload["data"] if 'data' in payload else None     # maybe use {} instead
    errors = payload["errors"] if 'errors' in payload else []

    if len(data):
        resp["data"] = data
        status = "ok"         # might just make this an dict keyed by status_code

    elif len(errors):
        resp["errors"] = errors
        if status_code == 400:
            status = "bad request"
        elif status_code in [401, 403]:
            status = "unauthorized"
        elif status_code == 404:
            status = "not found"          # dict keyed by status_code would be better

    else:
        resp["errors"] = [{"code": "something-went-wrong"}] # will this handle 500 errors?

    resp = {
        "status": status,
        "apiVersion": API_VERSION
        "data": data        # always include data even if its just the original POST form
        "errors" errors     # always include errors even if there are none
    }

    return jsonify(resp), status_code
```

Then the call would just be simple
```python
# success
standardize_response(data)

# failure
standardize_response(data, 404)
```

Not sure if this is what you had in mind but it's one option, another is to pass data and errors in as a named tuple.

**Comment by aaron-suarez** [comment]
You know, I forgot that keyword arguments are a thing in python, I may think about doing it that way... Thanks for the suggestion, I'm sure it will ultimately influence the final implementation.

If you wanted to open up a PR to fix it, that's fine too! Contributors wanted 😄

**Comment by aaron-suarez** [comment]
Closed in #110
